src/Adaptivity/AdaptedTriangulations.jl

```diff
@@ -0,0 +1,316 @@
+
+"""
+  AdaptedTriangulation
+
+  Triangulation produced from an AdaptedDiscreteModel.
+  
+  Contains: 
+
+  - adapted_model ~> AdaptedDiscreteModel for the triangulation.
+  - trian ~> Triangulation for the background model, i.e get_model(adapted_model).
+"""
+struct AdaptedTriangulation{Dc,Dp,A<:Triangulation{Dc,Dp},B<:AdaptedDiscreteModel} <: Triangulation{Dc,Dp}
+  trian::A
+  adapted_model::B
+
+  function AdaptedTriangulation(trian::Triangulation{Dc,Dp},model::AdaptedDiscreteModel{Dc2,Dp}) where {Dc,Dc2,Dp}
+    @check !isa(trian,AdaptedTriangulation)
+    @check Dc <= Dc2
```

Comment:
Should we also check here that  `get_background_model(trian)===get_model(model)` ?